Undefined reference in unit tests

Current version gives the below build error when building MainTest.  Maybe CMake must check whether CppUnit is installed or so?

<pre>
Linking CXX executable MainTest
CMakeFiles/MainTest.dir/ArcTest.cpp.o: In function `AutoRegisterSuite':
/usr/include/cppunit/extensions/AutoRegisterSuite.h:34: undefined reference to `CppUnit::TestFactoryRegistry::getRegistry(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
CMakeFiles/MainTest.dir/ArcTest.cpp.o: In function `ArcTest::suite()':
/tmp/oaconvert/test/ArcTest.h:30: undefined reference to `CppUnit::TestSuite::TestSuite(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >)'
CMakeFiles/MainTest.dir/ArcTest.cpp.o: In function `ArcTest::addTestsToSuite(CppUnit::TestSuiteBuilderContextBase&)':
/tmp/oaconvert/test/ArcTest.h:29: undefined reference to `CppUnit::TestSuiteBuilderContextBase::getTestNameFor(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) const'
CMakeFiles/MainTest.dir/ArcTest.cpp.o: In function `TestCaller':
/usr/include/cppunit/TestCaller.h:154: undefined reference to `CppUnit::TestCase::TestCase(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
CMakeFiles/MainTest.dir/ArcTest.cpp.o:(.rodata._ZTVN7CppUnit10TestCallerI7ArcTestEE[_ZTVN7CppUnit10TestCallerI7ArcTestEE]+0x24): undefined reference to `CppUnit::Test::findTestPath(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, CppUnit::TestPath&) const'
CMakeFiles/MainTest.dir/ArcTest.cpp.o:(.rodata._ZTVN7CppUnit10TestCallerI7ArcTestEE[_ZTVN7CppUnit10TestCallerI7ArcTestEE]+0x2c): undefined reference to `CppUnit::Test::findTest(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) const'
CMakeFiles/MainTest.dir/ArcTest.cpp.o:(.rodata._ZTVN7CppUnit10TestCallerI7ArcTestEE[_ZTVN7CppUnit10TestCallerI7ArcTestEE]+0x30): undefined reference to `CppUnit::Test::resolveTestPath(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) const'
CMakeFiles/MainTest.dir/LatLonTest.cpp.o: In function `LatLonTest::testGetAngle()':
/tmp/oaconvert/test/LatLonTest.cpp:43: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
/tmp/oaconvert/test/LatLonTest.cpp:43: undefined reference to `CppUnit::assertDoubleEquals(double, double, double, CppUnit::SourceLine, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
/tmp/oaconvert/test/LatLonTest.cpp:46: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
/tmp/oaconvert/test/LatLonTest.cpp:46: undefined reference to `CppUnit::assertDoubleEquals(double, double, double, CppUnit::SourceLine, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
/tmp/oaconvert/test/LatLonTest.cpp:49: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
/tmp/oaconvert/test/LatLonTest.cpp:49: undefined reference to `CppUnit::assertDoubleEquals(double, double, double, CppUnit::SourceLine, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
/tmp/oaconvert/test/LatLonTest.cpp:52: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
/tmp/oaconvert/test/LatLonTest.cpp:52: undefined reference to `CppUnit::assertDoubleEquals(double, double, double, CppUnit::SourceLine, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
/tmp/oaconvert/test/LatLonTest.cpp:55: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
/tmp/oaconvert/test/LatLonTest.cpp:55: undefined reference to `CppUnit::assertDoubleEquals(double, double, double, CppUnit::SourceLine, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
/tmp/oaconvert/test/LatLonTest.cpp:58: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
/tmp/oaconvert/test/LatLonTest.cpp:58: undefined reference to `CppUnit::assertDoubleEquals(double, double, double, CppUnit::SourceLine, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
/tmp/oaconvert/test/LatLonTest.cpp:61: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
/tmp/oaconvert/test/LatLonTest.cpp:61: undefined reference to `CppUnit::assertDoubleEquals(double, double, double, CppUnit::SourceLine, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
/tmp/oaconvert/test/LatLonTest.cpp:64: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
/tmp/oaconvert/test/LatLonTest.cpp:64: undefined reference to `CppUnit::assertDoubleEquals(double, double, double, CppUnit::SourceLine, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
CMakeFiles/MainTest.dir/LatLonTest.cpp.o: In function `LatLonTest::testGetDegrees()':
/tmp/oaconvert/test/LatLonTest.cpp:71: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
/tmp/oaconvert/test/LatLonTest.cpp:74: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
/tmp/oaconvert/test/LatLonTest.cpp:77: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
/tmp/oaconvert/test/LatLonTest.cpp:80: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
CMakeFiles/MainTest.dir/LatLonTest.cpp.o: In function `LatLonTest::testGetMinutes()':
/tmp/oaconvert/test/LatLonTest.cpp:87: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
CMakeFiles/MainTest.dir/LatLonTest.cpp.o:/tmp/oaconvert/test/LatLonTest.cpp:90: more undefined references to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)' follow
CMakeFiles/MainTest.dir/LatLonTest.cpp.o: In function `LatLonTest::testGetSeconds()':
/tmp/oaconvert/test/LatLonTest.cpp:105: undefined reference to `CppUnit::assertDoubleEquals(double, double, double, CppUnit::SourceLine, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
/tmp/oaconvert/test/LatLonTest.cpp:108: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
/tmp/oaconvert/test/LatLonTest.cpp:108: undefined reference to `CppUnit::assertDoubleEquals(double, double, double, CppUnit::SourceLine, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
/tmp/oaconvert/test/LatLonTest.cpp:111: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
/tmp/oaconvert/test/LatLonTest.cpp:111: undefined reference to `CppUnit::assertDoubleEquals(double, double, double, CppUnit::SourceLine, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
/tmp/oaconvert/test/LatLonTest.cpp:114: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
/tmp/oaconvert/test/LatLonTest.cpp:114: undefined reference to `CppUnit::assertDoubleEquals(double, double, double, CppUnit::SourceLine, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
CMakeFiles/MainTest.dir/LatLonTest.cpp.o: In function `AutoRegisterSuite':
/usr/include/cppunit/extensions/AutoRegisterSuite.h:34: undefined reference to `CppUnit::TestFactoryRegistry::getRegistry(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
CMakeFiles/MainTest.dir/LatLonTest.cpp.o: In function `void CppUnit::assertEquals<int>(int const&, int const&, CppUnit::SourceLine, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)':
/usr/include/cppunit/TestAssert.h:116: undefined reference to `CppUnit::AdditionalMessage::AdditionalMessage(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
/usr/include/cppunit/TestAssert.h:116: undefined reference to `CppUnit::Asserter::failNotEqual(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, CppUnit::SourceLine const&, CppUnit::AdditionalMessage const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >)'
CMakeFiles/MainTest.dir/LatLonTest.cpp.o: In function `LatLonTest::suite()':
/tmp/oaconvert/test/LatLonTest.h:33: undefined reference to `CppUnit::TestSuite::TestSuite(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >)'
CMakeFiles/MainTest.dir/LatLonTest.cpp.o: In function `LatLonTest::addTestsToSuite(CppUnit::TestSuiteBuilderContextBase&)':
/tmp/oaconvert/test/LatLonTest.h:29: undefined reference to `CppUnit::TestSuiteBuilderContextBase::getTestNameFor(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) const'
/tmp/oaconvert/test/LatLonTest.h:30: undefined reference to `CppUnit::TestSuiteBuilderContextBase::getTestNameFor(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) const'
/tmp/oaconvert/test/LatLonTest.h:31: undefined reference to `CppUnit::TestSuiteBuilderContextBase::getTestNameFor(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) const'
/tmp/oaconvert/test/LatLonTest.h:32: undefined reference to `CppUnit::TestSuiteBuilderContextBase::getTestNameFor(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) const'
CMakeFiles/MainTest.dir/LatLonTest.cpp.o: In function `TestCaller':
/usr/include/cppunit/TestCaller.h:154: undefined reference to `CppUnit::TestCase::TestCase(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
CMakeFiles/MainTest.dir/LatLonTest.cpp.o:(.rodata._ZTVN7CppUnit10TestCallerI10LatLonTestEE[_ZTVN7CppUnit10TestCallerI10LatLonTestEE]+0x24): undefined reference to `CppUnit::Test::findTestPath(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, CppUnit::TestPath&) const'
CMakeFiles/MainTest.dir/LatLonTest.cpp.o:(.rodata._ZTVN7CppUnit10TestCallerI10LatLonTestEE[_ZTVN7CppUnit10TestCallerI10LatLonTestEE]+0x2c): undefined reference to `CppUnit::Test::findTest(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) const'
CMakeFiles/MainTest.dir/LatLonTest.cpp.o:(.rodata._ZTVN7CppUnit10TestCallerI10LatLonTestEE[_ZTVN7CppUnit10TestCallerI10LatLonTestEE]+0x30): undefined reference to `CppUnit::Test::resolveTestPath(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) const'
CMakeFiles/MainTest.dir/LatitudeTest.cpp.o: In function `LatitudeTest::testConstructors()':
/tmp/oaconvert/test/LatitudeTest.cpp:40: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
/tmp/oaconvert/test/LatitudeTest.cpp:43: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
/tmp/oaconvert/test/LatitudeTest.cpp:47: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
/tmp/oaconvert/test/LatitudeTest.cpp:50: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
/tmp/oaconvert/test/LatitudeTest.cpp:53: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
CMakeFiles/MainTest.dir/LatitudeTest.cpp.o:/tmp/oaconvert/test/LatitudeTest.cpp:56: more undefined references to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)' follow
CMakeFiles/MainTest.dir/LatitudeTest.cpp.o: In function `LatitudeTest::testGetArcDegree()':
/tmp/oaconvert/test/LatitudeTest.cpp:116: undefined reference to `CppUnit::assertDoubleEquals(double, double, double, CppUnit::SourceLine, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
/tmp/oaconvert/test/LatitudeTest.cpp:119: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
/tmp/oaconvert/test/LatitudeTest.cpp:119: undefined reference to `CppUnit::assertDoubleEquals(double, double, double, CppUnit::SourceLine, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
/tmp/oaconvert/test/LatitudeTest.cpp:122: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
/tmp/oaconvert/test/LatitudeTest.cpp:122: undefined reference to `CppUnit::assertDoubleEquals(double, double, double, CppUnit::SourceLine, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
CMakeFiles/MainTest.dir/LatitudeTest.cpp.o: In function `AutoRegisterSuite':
/usr/include/cppunit/extensions/AutoRegisterSuite.h:34: undefined reference to `CppUnit::TestFactoryRegistry::getRegistry(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
CMakeFiles/MainTest.dir/LatitudeTest.cpp.o: In function `void CppUnit::assertEquals<char>(char const&, char const&, CppUnit::SourceLine, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)':
/usr/include/cppunit/TestAssert.h:116: undefined reference to `CppUnit::AdditionalMessage::AdditionalMessage(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
/usr/include/cppunit/TestAssert.h:116: undefined reference to `CppUnit::Asserter::failNotEqual(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, CppUnit::SourceLine const&, CppUnit::AdditionalMessage const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >)'
CMakeFiles/MainTest.dir/LatitudeTest.cpp.o: In function `LatitudeTest::suite()':
/tmp/oaconvert/test/LatitudeTest.h:32: undefined reference to `CppUnit::TestSuite::TestSuite(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >)'
CMakeFiles/MainTest.dir/LatitudeTest.cpp.o: In function `LatitudeTest::addTestsToSuite(CppUnit::TestSuiteBuilderContextBase&)':
/tmp/oaconvert/test/LatitudeTest.h:29: undefined reference to `CppUnit::TestSuiteBuilderContextBase::getTestNameFor(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) const'
/tmp/oaconvert/test/LatitudeTest.h:30: undefined reference to `CppUnit::TestSuiteBuilderContextBase::getTestNameFor(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) const'
/tmp/oaconvert/test/LatitudeTest.h:31: undefined reference to `CppUnit::TestSuiteBuilderContextBase::getTestNameFor(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) const'
CMakeFiles/MainTest.dir/LatitudeTest.cpp.o: In function `TestCaller':
/usr/include/cppunit/TestCaller.h:154: undefined reference to `CppUnit::TestCase::TestCase(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
CMakeFiles/MainTest.dir/LatitudeTest.cpp.o:(.rodata._ZTVN7CppUnit10TestCallerI12LatitudeTestEE[_ZTVN7CppUnit10TestCallerI12LatitudeTestEE]+0x24): undefined reference to `CppUnit::Test::findTestPath(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, CppUnit::TestPath&) const'
CMakeFiles/MainTest.dir/LatitudeTest.cpp.o:(.rodata._ZTVN7CppUnit10TestCallerI12LatitudeTestEE[_ZTVN7CppUnit10TestCallerI12LatitudeTestEE]+0x2c): undefined reference to `CppUnit::Test::findTest(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) const'
CMakeFiles/MainTest.dir/LatitudeTest.cpp.o:(.rodata._ZTVN7CppUnit10TestCallerI12LatitudeTestEE[_ZTVN7CppUnit10TestCallerI12LatitudeTestEE]+0x30): undefined reference to `CppUnit::Test::resolveTestPath(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) const'
CMakeFiles/MainTest.dir/LongitudeTest.cpp.o: In function `LongitudeTest::testConstructors()':
/tmp/oaconvert/test/LongitudeTest.cpp:41: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
/tmp/oaconvert/test/LongitudeTest.cpp:44: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
/tmp/oaconvert/test/LongitudeTest.cpp:48: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
/tmp/oaconvert/test/LongitudeTest.cpp:51: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
/tmp/oaconvert/test/LongitudeTest.cpp:54: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
CMakeFiles/MainTest.dir/LongitudeTest.cpp.o:/tmp/oaconvert/test/LongitudeTest.cpp:57: more undefined references to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)' follow
CMakeFiles/MainTest.dir/LongitudeTest.cpp.o: In function `LongitudeTest::testGetArcDegree()':
/tmp/oaconvert/test/LongitudeTest.cpp:101: undefined reference to `CppUnit::assertDoubleEquals(double, double, double, CppUnit::SourceLine, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
/tmp/oaconvert/test/LongitudeTest.cpp:104: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
/tmp/oaconvert/test/LongitudeTest.cpp:104: undefined reference to `CppUnit::assertDoubleEquals(double, double, double, CppUnit::SourceLine, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
/tmp/oaconvert/test/LongitudeTest.cpp:107: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
/tmp/oaconvert/test/LongitudeTest.cpp:107: undefined reference to `CppUnit::assertDoubleEquals(double, double, double, CppUnit::SourceLine, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
CMakeFiles/MainTest.dir/LongitudeTest.cpp.o: In function `AutoRegisterSuite':
/usr/include/cppunit/extensions/AutoRegisterSuite.h:34: undefined reference to `CppUnit::TestFactoryRegistry::getRegistry(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
CMakeFiles/MainTest.dir/LongitudeTest.cpp.o: In function `LongitudeTest::suite()':
/tmp/oaconvert/test/LongitudeTest.h:31: undefined reference to `CppUnit::TestSuite::TestSuite(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >)'
CMakeFiles/MainTest.dir/LongitudeTest.cpp.o: In function `LongitudeTest::addTestsToSuite(CppUnit::TestSuiteBuilderContextBase&)':
/tmp/oaconvert/test/LongitudeTest.h:29: undefined reference to `CppUnit::TestSuiteBuilderContextBase::getTestNameFor(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) const'
/tmp/oaconvert/test/LongitudeTest.h:30: undefined reference to `CppUnit::TestSuiteBuilderContextBase::getTestNameFor(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) const'
CMakeFiles/MainTest.dir/LongitudeTest.cpp.o: In function `TestCaller':
/usr/include/cppunit/TestCaller.h:154: undefined reference to `CppUnit::TestCase::TestCase(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
CMakeFiles/MainTest.dir/LongitudeTest.cpp.o:(.rodata._ZTVN7CppUnit10TestCallerI13LongitudeTestEE[_ZTVN7CppUnit10TestCallerI13LongitudeTestEE]+0x24): undefined reference to `CppUnit::Test::findTestPath(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, CppUnit::TestPath&) const'
CMakeFiles/MainTest.dir/LongitudeTest.cpp.o:(.rodata._ZTVN7CppUnit10TestCallerI13LongitudeTestEE[_ZTVN7CppUnit10TestCallerI13LongitudeTestEE]+0x2c): undefined reference to `CppUnit::Test::findTest(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) const'
CMakeFiles/MainTest.dir/LongitudeTest.cpp.o:(.rodata._ZTVN7CppUnit10TestCallerI13LongitudeTestEE[_ZTVN7CppUnit10TestCallerI13LongitudeTestEE]+0x30): undefined reference to `CppUnit::Test::resolveTestPath(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) const'
CMakeFiles/MainTest.dir/CoordinateTest.cpp.o: In function `CoordinateTest::testGetDistance()':
/tmp/oaconvert/test/CoordinateTest.cpp:48: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
/tmp/oaconvert/test/CoordinateTest.cpp:48: undefined reference to `CppUnit::assertDoubleEquals(double, double, double, CppUnit::SourceLine, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
CMakeFiles/MainTest.dir/CoordinateTest.cpp.o: In function `AutoRegisterSuite':
/usr/include/cppunit/extensions/AutoRegisterSuite.h:34: undefined reference to `CppUnit::TestFactoryRegistry::getRegistry(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
CMakeFiles/MainTest.dir/CoordinateTest.cpp.o: In function `CoordinateTest::suite()':
/tmp/oaconvert/test/CoordinateTest.h:31: undefined reference to `CppUnit::TestSuite::TestSuite(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >)'
CMakeFiles/MainTest.dir/CoordinateTest.cpp.o: In function `CoordinateTest::addTestsToSuite(CppUnit::TestSuiteBuilderContextBase&)':
/tmp/oaconvert/test/CoordinateTest.h:29: undefined reference to `CppUnit::TestSuiteBuilderContextBase::getTestNameFor(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) const'
/tmp/oaconvert/test/CoordinateTest.h:30: undefined reference to `CppUnit::TestSuiteBuilderContextBase::getTestNameFor(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) const'
CMakeFiles/MainTest.dir/CoordinateTest.cpp.o: In function `TestCaller':
/usr/include/cppunit/TestCaller.h:154: undefined reference to `CppUnit::TestCase::TestCase(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
CMakeFiles/MainTest.dir/CoordinateTest.cpp.o:(.rodata._ZTVN7CppUnit10TestCallerI14CoordinateTestEE[_ZTVN7CppUnit10TestCallerI14CoordinateTestEE]+0x24): undefined reference to `CppUnit::Test::findTestPath(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, CppUnit::TestPath&) const'
CMakeFiles/MainTest.dir/CoordinateTest.cpp.o:(.rodata._ZTVN7CppUnit10TestCallerI14CoordinateTestEE[_ZTVN7CppUnit10TestCallerI14CoordinateTestEE]+0x2c): undefined reference to `CppUnit::Test::findTest(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) const'
CMakeFiles/MainTest.dir/CoordinateTest.cpp.o:(.rodata._ZTVN7CppUnit10TestCallerI14CoordinateTestEE[_ZTVN7CppUnit10TestCallerI14CoordinateTestEE]+0x30): undefined reference to `CppUnit::Test::resolveTestPath(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) const'
CMakeFiles/MainTest.dir/ParserTest.cpp.o: In function `ParserTest::testParseCoordinate()':
/tmp/oaconvert/test/ParserTest.cpp:49: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
/tmp/oaconvert/test/ParserTest.cpp:55: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
/tmp/oaconvert/test/ParserTest.cpp:61: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
/tmp/oaconvert/test/ParserTest.cpp:67: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
CMakeFiles/MainTest.dir/ParserTest.cpp.o: In function `ParserTest::testParseAltitude()':
/tmp/oaconvert/test/ParserTest.cpp:104: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
/tmp/oaconvert/test/ParserTest.cpp:104: undefined reference to `CppUnit::assertDoubleEquals(double, double, double, CppUnit::SourceLine, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
/tmp/oaconvert/test/ParserTest.cpp:107: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
/tmp/oaconvert/test/ParserTest.cpp:107: undefined reference to `CppUnit::assertDoubleEquals(double, double, double, CppUnit::SourceLine, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
/tmp/oaconvert/test/ParserTest.cpp:110: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
/tmp/oaconvert/test/ParserTest.cpp:110: undefined reference to `CppUnit::assertDoubleEquals(double, double, double, CppUnit::SourceLine, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
/tmp/oaconvert/test/ParserTest.cpp:113: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
/tmp/oaconvert/test/ParserTest.cpp:113: undefined reference to `CppUnit::assertDoubleEquals(double, double, double, CppUnit::SourceLine, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
/tmp/oaconvert/test/ParserTest.cpp:116: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
/tmp/oaconvert/test/ParserTest.cpp:116: undefined reference to `CppUnit::assertDoubleEquals(double, double, double, CppUnit::SourceLine, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
/tmp/oaconvert/test/ParserTest.cpp:119: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
/tmp/oaconvert/test/ParserTest.cpp:119: undefined reference to `CppUnit::assertDoubleEquals(double, double, double, CppUnit::SourceLine, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
/tmp/oaconvert/test/ParserTest.cpp:122: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
/tmp/oaconvert/test/ParserTest.cpp:122: undefined reference to `CppUnit::assertDoubleEquals(double, double, double, CppUnit::SourceLine, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
/tmp/oaconvert/test/ParserTest.cpp:125: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
/tmp/oaconvert/test/ParserTest.cpp:125: undefined reference to `CppUnit::assertDoubleEquals(double, double, double, CppUnit::SourceLine, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
/tmp/oaconvert/test/ParserTest.cpp:128: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
/tmp/oaconvert/test/ParserTest.cpp:128: undefined reference to `CppUnit::assertDoubleEquals(double, double, double, CppUnit::SourceLine, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
/tmp/oaconvert/test/ParserTest.cpp:131: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
/tmp/oaconvert/test/ParserTest.cpp:131: undefined reference to `CppUnit::assertDoubleEquals(double, double, double, CppUnit::SourceLine, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
/tmp/oaconvert/test/ParserTest.cpp:134: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
/tmp/oaconvert/test/ParserTest.cpp:134: undefined reference to `CppUnit::assertDoubleEquals(double, double, double, CppUnit::SourceLine, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
/tmp/oaconvert/test/ParserTest.cpp:137: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
/tmp/oaconvert/test/ParserTest.cpp:137: undefined reference to `CppUnit::assertDoubleEquals(double, double, double, CppUnit::SourceLine, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
/tmp/oaconvert/test/ParserTest.cpp:140: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
/tmp/oaconvert/test/ParserTest.cpp:140: undefined reference to `CppUnit::assertDoubleEquals(double, double, double, CppUnit::SourceLine, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
/tmp/oaconvert/test/ParserTest.cpp:143: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
/tmp/oaconvert/test/ParserTest.cpp:143: undefined reference to `CppUnit::assertDoubleEquals(double, double, double, CppUnit::SourceLine, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
/tmp/oaconvert/test/ParserTest.cpp:146: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
/tmp/oaconvert/test/ParserTest.cpp:146: undefined reference to `CppUnit::assertDoubleEquals(double, double, double, CppUnit::SourceLine, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
CMakeFiles/MainTest.dir/ParserTest.cpp.o: In function `ParserTest::testParseAirspaceClass()':
/tmp/oaconvert/test/ParserTest.cpp:160: undefined reference to `CppUnit::Message::Message(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
/tmp/oaconvert/test/ParserTest.cpp:160: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
/tmp/oaconvert/test/ParserTest.cpp:163: undefined reference to `CppUnit::Message::Message(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
/tmp/oaconvert/test/ParserTest.cpp:163: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
/tmp/oaconvert/test/ParserTest.cpp:166: undefined reference to `CppUnit::Message::Message(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
/tmp/oaconvert/test/ParserTest.cpp:166: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
/tmp/oaconvert/test/ParserTest.cpp:169: undefined reference to `CppUnit::Message::Message(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
/tmp/oaconvert/test/ParserTest.cpp:169: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
/tmp/oaconvert/test/ParserTest.cpp:172: undefined reference to `CppUnit::Message::Message(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
/tmp/oaconvert/test/ParserTest.cpp:172: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
/tmp/oaconvert/test/ParserTest.cpp:175: undefined reference to `CppUnit::Message::Message(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
/tmp/oaconvert/test/ParserTest.cpp:175: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
/tmp/oaconvert/test/ParserTest.cpp:178: undefined reference to `CppUnit::Message::Message(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
/tmp/oaconvert/test/ParserTest.cpp:178: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
/tmp/oaconvert/test/ParserTest.cpp:181: undefined reference to `CppUnit::Message::Message(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
/tmp/oaconvert/test/ParserTest.cpp:181: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
/tmp/oaconvert/test/ParserTest.cpp:184: undefined reference to `CppUnit::Message::Message(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
/tmp/oaconvert/test/ParserTest.cpp:184: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
/tmp/oaconvert/test/ParserTest.cpp:187: undefined reference to `CppUnit::Message::Message(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
/tmp/oaconvert/test/ParserTest.cpp:187: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
/tmp/oaconvert/test/ParserTest.cpp:190: undefined reference to `CppUnit::Message::Message(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
/tmp/oaconvert/test/ParserTest.cpp:190: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
/tmp/oaconvert/test/ParserTest.cpp:193: undefined reference to `CppUnit::Message::Message(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
/tmp/oaconvert/test/ParserTest.cpp:193: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
/tmp/oaconvert/test/ParserTest.cpp:196: undefined reference to `CppUnit::Message::Message(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
/tmp/oaconvert/test/ParserTest.cpp:196: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
/tmp/oaconvert/test/ParserTest.cpp:199: undefined reference to `CppUnit::Message::Message(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
/tmp/oaconvert/test/ParserTest.cpp:199: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
/tmp/oaconvert/test/ParserTest.cpp:202: undefined reference to `CppUnit::Message::Message(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
/tmp/oaconvert/test/ParserTest.cpp:202: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
CMakeFiles/MainTest.dir/ParserTest.cpp.o: In function `AutoRegisterSuite':
/usr/include/cppunit/extensions/AutoRegisterSuite.h:34: undefined reference to `CppUnit::TestFactoryRegistry::getRegistry(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
CMakeFiles/MainTest.dir/ParserTest.cpp.o: In function `void CppUnit::assertEquals<Coordinate>(Coordinate const&, Coordinate const&, CppUnit::SourceLine, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)':
/usr/include/cppunit/TestAssert.h:116: undefined reference to `CppUnit::AdditionalMessage::AdditionalMessage(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
/usr/include/cppunit/TestAssert.h:116: undefined reference to `CppUnit::Asserter::failNotEqual(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, CppUnit::SourceLine const&, CppUnit::AdditionalMessage const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >)'
CMakeFiles/MainTest.dir/ParserTest.cpp.o: In function `ParserTest::suite()':
/tmp/oaconvert/test/ParserTest.h:32: undefined reference to `CppUnit::TestSuite::TestSuite(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >)'
CMakeFiles/MainTest.dir/ParserTest.cpp.o: In function `ParserTest::addTestsToSuite(CppUnit::TestSuiteBuilderContextBase&)':
/tmp/oaconvert/test/ParserTest.h:29: undefined reference to `CppUnit::TestSuiteBuilderContextBase::getTestNameFor(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) const'
/tmp/oaconvert/test/ParserTest.h:30: undefined reference to `CppUnit::TestSuiteBuilderContextBase::getTestNameFor(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) const'
/tmp/oaconvert/test/ParserTest.h:31: undefined reference to `CppUnit::TestSuiteBuilderContextBase::getTestNameFor(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) const'
CMakeFiles/MainTest.dir/ParserTest.cpp.o: In function `TestCaller':
/usr/include/cppunit/TestCaller.h:154: undefined reference to `CppUnit::TestCase::TestCase(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
CMakeFiles/MainTest.dir/ParserTest.cpp.o:(.rodata._ZTVN7CppUnit10TestCallerI10ParserTestEE[_ZTVN7CppUnit10TestCallerI10ParserTestEE]+0x24): undefined reference to `CppUnit::Test::findTestPath(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, CppUnit::TestPath&) const'
CMakeFiles/MainTest.dir/ParserTest.cpp.o:(.rodata._ZTVN7CppUnit10TestCallerI10ParserTestEE[_ZTVN7CppUnit10TestCallerI10ParserTestEE]+0x2c): undefined reference to `CppUnit::Test::findTest(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) const'
CMakeFiles/MainTest.dir/ParserTest.cpp.o:(.rodata._ZTVN7CppUnit10TestCallerI10ParserTestEE[_ZTVN7CppUnit10TestCallerI10ParserTestEE]+0x30): undefined reference to `CppUnit::Test::resolveTestPath(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) const'
CMakeFiles/MainTest.dir/AirspaceTest.cpp.o: In function `AirspaceTest::testSetGetName()':
/tmp/oaconvert/test/AirspaceTest.cpp:44: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
CMakeFiles/MainTest.dir/AirspaceTest.cpp.o: In function `AirspaceTest::testComparison()':
/tmp/oaconvert/test/AirspaceTest.cpp:68: undefined reference to `CppUnit::Message::Message(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
/tmp/oaconvert/test/AirspaceTest.cpp:68: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
/tmp/oaconvert/test/AirspaceTest.cpp:72: undefined reference to `CppUnit::Message::Message(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
/tmp/oaconvert/test/AirspaceTest.cpp:72: undefined reference to `CppUnit::SourceLine::SourceLine(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)'
CMakeFiles/MainTest.dir/AirspaceTest.cpp.o: In function `AutoRegisterSuite':
/usr/include/cppunit/extensions/AutoRegisterSuite.h:34: undefined reference to `CppUnit::TestFactoryRegistry::getRegistry(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
CMakeFiles/MainTest.dir/AirspaceTest.cpp.o: In function `void CppUnit::assertEquals<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, CppUnit::SourceLine, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)':
/usr/include/cppunit/TestAssert.h:116: undefined reference to `CppUnit::AdditionalMessage::AdditionalMessage(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
/usr/include/cppunit/TestAssert.h:116: undefined reference to `CppUnit::Asserter::failNotEqual(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, CppUnit::SourceLine const&, CppUnit::AdditionalMessage const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >)'
CMakeFiles/MainTest.dir/AirspaceTest.cpp.o: In function `AirspaceTest::suite()':
/tmp/oaconvert/test/AirspaceTest.h:32: undefined reference to `CppUnit::TestSuite::TestSuite(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >)'
CMakeFiles/MainTest.dir/AirspaceTest.cpp.o: In function `AirspaceTest::addTestsToSuite(CppUnit::TestSuiteBuilderContextBase&)':
/tmp/oaconvert/test/AirspaceTest.h:29: undefined reference to `CppUnit::TestSuiteBuilderContextBase::getTestNameFor(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) const'
/tmp/oaconvert/test/AirspaceTest.h:30: undefined reference to `CppUnit::TestSuiteBuilderContextBase::getTestNameFor(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) const'
/tmp/oaconvert/test/AirspaceTest.h:31: undefined reference to `CppUnit::TestSuiteBuilderContextBase::getTestNameFor(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) const'
CMakeFiles/MainTest.dir/AirspaceTest.cpp.o: In function `TestCaller':
/usr/include/cppunit/TestCaller.h:154: undefined reference to `CppUnit::TestCase::TestCase(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
CMakeFiles/MainTest.dir/AirspaceTest.cpp.o:(.rodata._ZTVN7CppUnit10TestCallerI12AirspaceTestEE[_ZTVN7CppUnit10TestCallerI12AirspaceTestEE]+0x24): undefined reference to `CppUnit::Test::findTestPath(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, CppUnit::TestPath&) const'
CMakeFiles/MainTest.dir/AirspaceTest.cpp.o:(.rodata._ZTVN7CppUnit10TestCallerI12AirspaceTestEE[_ZTVN7CppUnit10TestCallerI12AirspaceTestEE]+0x2c): undefined reference to `CppUnit::Test::findTest(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) const'
CMakeFiles/MainTest.dir/AirspaceTest.cpp.o:(.rodata._ZTVN7CppUnit10TestCallerI12AirspaceTestEE[_ZTVN7CppUnit10TestCallerI12AirspaceTestEE]+0x30): undefined reference to `CppUnit::Test::resolveTestPath(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) const'
CMakeFiles/MainTest.dir/MainTester.cpp.o: In function `main':
/tmp/oaconvert/test/MainTester.cpp:42: undefined reference to `CppUnit::TestFactoryRegistry::getRegistry(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
/tmp/oaconvert/test/MainTester.cpp:43: undefined reference to `CppUnit::TestRunner::run(CppUnit::TestResult&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
/tmp/oaconvert/test/MainTester.cpp:46: undefined reference to `CppUnit::CompilerOutputter::CompilerOutputter(CppUnit::TestResultCollector*, std::__1::basic_ostream<char, std::__1::char_traits<char> >&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)'
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [test/MainTest] Error 1
make[1]: *** [test/CMakeFiles/MainTest.dir/all] Error 2
make: *** [all] Error 2
</pre>

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -12,8 +12,7 @@ find_program(ASTYLE astyle)
 
 
 if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU" OR "${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -g")
-    set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -Wall -Wextra -Weffc++ -Wold-style-cast -Woverloaded-virtual -Wsign-promo -Wabi -Wmissing-declarations -Wconversion -Wcast-align -Wredundant-decls -Werror -pedantic")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -Wall -Wextra -Weffc++ -Wold-style-cast -Woverloaded-virtual -Wsign-promo -Wabi -Wmissing-declarations -Wconversion -Wcast-align -Wredundant-decls -Werror -pedantic")
 endif()
 
 if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -8,6 +8,9 @@ cmake_minimum_required(VERSION 2.6)
 
 project(oaconvert CXX)
 
+find_program(ASTYLE astyle)
+
+
 if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU" OR "${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -g")
     set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -Wall -Wextra -Weffc++ -Wold-style-cast -Woverloaded-virtual -Wsign-promo -Wabi -Wmissing-declarations -Wconversion -Wcast-align -Wredundant-decls -Werror -pedantic")
@@ -25,17 +28,19 @@ endif()
 add_subdirectory(src)
 add_subdirectory(test)
 
-add_custom_target(reformat
-                  astyle --options=astyle_options src/*.cpp src/*.h
-                  COMMENT "Beautifying CPP source files"
-                  VERBATIM)
+if(ASTYLE)
+    add_custom_target(reformat
+                      ${ASTYLE} --options=astyle_options src/*.cpp src/*.h 
+                      COMMENT "Beautifying CPP source files"
+                      WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+                      VERBATIM)
 
-# TODO: this is not working yet, fix it.  The problem is
-#       the quotes around "*.orig" when typing `make origclean`.
-#add_custom_target(origclean
-#                  rm -f *.orig
-#                  COMMENT "Removing astyle backup files"
-#                  WORKING_DIRECTORY src
-#                  VERBATIM)
+    add_custom_target(origclean
+                      rm -f *.orig
+                      COMMENT "Removing astyle backup files"
+                      WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/src)
+else()
+    message(WARNING "Could not find astyle. To be able to beautify the source code, install it from http://astyle.sourceforge.net/.")
+endif()
 
 # vim: expandtab

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -9,6 +9,8 @@ cmake_minimum_required(VERSION 2.6)
 project(oaconvert CXX)
 
 find_program(ASTYLE astyle)
+find_library(CPPUNIT LIBRARY
+    NAMES cppunit)
 
 
 if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU" OR "${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
@@ -25,7 +27,15 @@ elseif("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
 endif()
 
 add_subdirectory(src)
-add_subdirectory(test)
+
+# Only compile the unit tests if cppunit is available or if we know that we can
+# compile it ourselves.
+if(CPPUNIT OR "${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
+  add_subdirectory(test)
+else()
+  message(WARNING "Cppunit was not found and we could not compile it "
+                  "automatically ourselves. Unit testing will be disabled.")
+endif()
 
 if(ASTYLE)
     add_custom_target(reformat

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -12,7 +12,7 @@ find_program(ASTYLE astyle)
 
 
 if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU" OR "${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -Wall -Wextra -Weffc++ -Wold-style-cast -Woverloaded-virtual -Wsign-promo -Wabi -Wmissing-declarations -Wconversion -Wcast-align -Wredundant-decls -Werror -pedantic")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -Wall -Wextra -Weffc++ -Wold-style-cast -Woverloaded-virtual -Wsign-promo -Wabi -Wmissing-declarations -Wconversion -Wcast-align -Wredundant-decls -pedantic")
 endif()
 
 if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -6,13 +6,21 @@
 set(CMAKE_LEGACY_CYGWIN_WIN32 0)
 cmake_minimum_required(VERSION 2.6)
 
-project(oaconvert)
+project(oaconvert CXX)
 
-# Uncomment this to use clang for compiling.
-SET (CMAKE_CXX_COMPILER "/usr/bin/clang++-3.5")
+if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU" OR "${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -g")
+    set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -Wall -Wextra -Weffc++ -Wold-style-cast -Woverloaded-virtual -Wsign-promo -Wabi -Wmissing-declarations -Wconversion -Wcast-align -Wredundant-decls -Werror -pedantic")
+endif()
 
-set(CMAKE_CXX_FLAGS "-std=c++14 -stdlib=libc++ -g")
-set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -Wall -Wextra -Weffc++ -Wold-style-cast -Woverloaded-virtual -Wsign-promo -Wabi -Wmissing-declarations -Wconversion -Wcast-align -Wredundant-decls -Werror -pedantic")
+if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
+    set(GCC_MIN_VER "4.9")
+    if(${CMAKE_CXX_COMPILER_VERSION} VERSION_LESS "${GCC_MIN_VER}")
+        message(FATAL_ERROR "GCC version should be at least ${GCC_MIN_VER}!")
+    endif()
+elseif("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -stdlib=libc++")
+endif()
 
 add_subdirectory(src)
 add_subdirectory(test)
@@ -29,3 +37,5 @@ add_custom_target(reformat
 #                  COMMENT "Removing astyle backup files"
 #                  WORKING_DIRECTORY src
 #                  VERBATIM)
+
+# vim: expandtab

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -28,14 +28,7 @@ endif()
 
 add_subdirectory(src)
 
-# Only compile the unit tests if cppunit is available or if we know that we can
-# compile it ourselves.
-if(CPPUNIT OR "${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
-  add_subdirectory(test)
-else()
-  message(WARNING "Cppunit was not found and we could not compile it "
-                  "automatically ourselves. Unit testing will be disabled.")
-endif()
+add_subdirectory(test)
 
 if(ASTYLE)
     add_custom_target(reformat

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -21,6 +21,6 @@ set(oaconvert_SRCS
     oaconvert.cpp
 )
 
-add_library(airspace SHARED ${airspace_SRCS})
+add_library(airspace STATIC ${airspace_SRCS})
 add_executable(oaconvert ${oaconvert_SRCS})
 target_link_libraries(oaconvert airspace)

--- a/src/oaconvert.cpp
+++ b/src/oaconvert.cpp
@@ -19,134 +19,128 @@
 
 #include <iostream>
 #include <fstream>
-#include <getopt.h>
-#include <strings.h>
+#include <string>
 
 #include "Airspace.h"
 #include "Latitude.h"
 #include "Parser.h"
-#include "StringUtils.h"
-
-using namespace std;
 
 int main (int argc, char* argv[])
 {
 
     if (argc < 2)
     {
-        fprintf(stderr,
-                "Usage: %s [-d|--debug level] [-o|--output polish_file.mp] open_air_file.txt\n",
-                argv[0]);
+        std::cerr << "Usage: " << argv[0] << " [-d|--debug level] [-o|--output polish_file.mp] open_air_file.txt" << std::endl;
         exit(EXIT_FAILURE);
     }
 
     // By default, if no output file is given, write to stdout.
-    bool write_to_stdout = true;
     std::string outfilename;
+    std::string infilename;
 
-    static struct option long_options[] =
+    auto getNextParameter = [argc, argv](int index) -> std::string
     {
-        {  "output", required_argument, 0, 'o'},
-        {   "units", required_argument, 0, 'u'},
-        {   "debug", optional_argument, 0, 'd'},
-        {    "help",       no_argument, 0, 'h'},
-        { "version",       no_argument, 0, 'v'},
-        {         0,                 0, 0,  0 }
+        if(index >= argc)
+        {
+            std::cerr << "ERROR: invalid index '" << index << "'. There are only " << argc << " arguments." << std::endl;
+        }
+        if(index + 1 < argc)
+        {
+            return argv[index + 1];
+        }
+        else
+        {
+            std::cerr << "ERROR: Expected argument for " << argv[index] << "." << std::endl;
+            exit(EXIT_FAILURE);
+        }
     };
 
-
-    int opt;
-    int option_index;
-
-    while ( (opt = getopt_long(argc, argv, "d::u:o:hv", long_options, &option_index)) != -1 )
+    for(int i = 1; i < argc; ++i)
     {
+        std::string option = argv[i];
 
-        switch (opt)
+        if(option == "-u" || option == "--units")
         {
-
-        case 0:
-            printf ("option %s", long_options[option_index].name);
-            if (optarg)
-                printf (" with arg %s", optarg);
-            printf ("\n");
-            break;
-
-        case 'u':
-
-            if (!strcasecmp(optarg, "ft"))   // this is counterintuitive...
+            auto units = getNextParameter(++i);
+            if(units == "ft")
             {
-                printf ("TODO: setting output display units to feet.\n");
+                std::cout << "TODO: setting output display units to feet." << std::endl;
             }
-            else if (!strcasecmp(optarg, "m"))     // this is counterintuitive...
+            else if(units == "m")
             {
-                printf ("TODO: setting output display units to meters.\n");
+                std::cout << "TODO: setting output display units to meters." << std::endl;
             }
             else
             {
-                printf("ERROR: display units must be 'ft' (feet) or 'm' (meters).\n");
-                exit(EXIT_FAILURE);
+                std::cout << "ERROR: display units must be 'ft' (feet) or 'm' (meters)." << std::endl;
+                return EXIT_FAILURE;
             }
-            break;
-
-        case 'o':
-            outfilename = optarg;
-            write_to_stdout = false;
-            break;
-
-        case 'd':
-            printf("Requested debug level: %s\n", optarg);
-            printf("WARNING: Debug level output is not implemented yet!\n");
-            break;
-
-        case 'h':
-            printf("TODO: implement help text.\n");
-            exit(EXIT_SUCCESS);
-
-        case 'v':
-            printf("oa2convert <some_version>\n");
-            printf("Copyright (C) 2011 Bart Vandewoestyne, Yves Frederix\n");
-            printf("License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>\n");
-            printf("This is free software: you are free to change and redistribute it.\n");
-            printf("There is NO WARRANTY, to the extent permitted by law.\n");
-            exit(EXIT_SUCCESS);
-
-        default: /* '?' */
-            fprintf(stderr,
-                    "Usage: %s [-d|--debug level] [-o|--output polish_file.mp] open_air_file.txt\n",
-                    argv[0]);
-            exit(EXIT_FAILURE);
         }
 
+        else if(option == "-o" || option == "--output")
+        {
+            outfilename = getNextParameter(i++);
+        }
+
+        else if(option == "-d" || option == "--debug")
+        {
+            auto debuglevel = getNextParameter(i++);
+            std::cout << "Requested debug level: " << debuglevel << std::endl;
+            std::cout << "WARNING: Debug level output is not implemented yet!" << std::endl;
+        }
+
+        else if(option == "-h" || option == "--help")
+        {
+            std::cout << "TODO: implement help text." << std::endl;
+            exit(EXIT_SUCCESS);
+        }
+
+        else if(option == "-v" || option == "--version")
+        {
+            std::cout << "oa2convert <some_version>" << std::endl;
+            std::cout << "Copyright (C) 2011 Bart Vandewoestyne, Yves Frederix" << std::endl;
+            std::cout << "License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>" << std::endl;
+            std::cout << "This is free software: you are free to change and redistribute it." << std::endl;
+            std::cout << "There is NO WARRANTY, to the extent permitted by law." << std::endl;
+            exit(EXIT_SUCCESS);
+        }
+
+        else
+        {
+            if(!infilename.empty())
+            {
+                std::cerr << "WARNING: more than one input file given. Ignoring '" << option << "'..." << std::endl;
+            }
+            else
+            {
+                infilename = option;
+            }
+        }
     }
 
-    // At least one non-option argument, i.e., the input file.
-    if (optind >= argc)
+    if (infilename.empty())
     {
-        fprintf(stderr, "ERROR: you must provide an OpenAir file after your options!!!\n");
+        std::cerr << "ERROR: you must provide an OpenAir file after your options!!!" << std::endl;
         exit(EXIT_FAILURE);
     }
 
-    // Read filename of the input file.
-    std::string infilename( argv[optind] );
-
-
     // Setup the parser
-    Parser *p;
+    std::unique_ptr<Parser> p;
 
-    if ( write_to_stdout )
+    if (outfilename.empty())
     {
-        p = new Parser();
+        p = std::unique_ptr<Parser>(new Parser());
     }
     else
     {
-        p = new Parser( outfilename );
+        p = std::unique_ptr<Parser>(new Parser(outfilename));
         //cout << "INFO: output file has extension ." << p->parseFileExtension( outfilename ) << "." << endl;
     }
 
     // Read the input (OpenAir) file.
     std::string infile(infilename);
-    string line;
-    ifstream inStream;
+    std::string line;
+    std::ifstream inStream;
     inStream.open(infile.c_str());
     if (inStream.is_open())
     {
@@ -160,14 +154,12 @@ int main (int argc, char* argv[])
     }
     else
     {
-        cerr << "ERROR: Unable to open file!\n";
+        std::cerr << "ERROR: Unable to open file '" << infilename << "'!" << std::endl;
+        exit(EXIT_FAILURE);
     }
 
     // Everything is parsed and stored, now write!
     p->writeAirspaces();
-
-    // cleanup
-    delete p;
 
     return 0;
 }

--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -1,3 +1,30 @@
+if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
+    # When building with Clang, we need to compile cppunit ourselves or we get
+    # linking errors...
+    include(ExternalProject)
+    ExternalProject_Add(
+        cppunit_external
+        CMAKE_ARGS -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
+        PREFIX ${CMAKE_BINARY_DIR}/cppunit
+        URL http://dev-www.libreoffice.org/src/cppunit-1.13.2.tar.gz
+        URL_MD5 d1c6bdd5a76c66d2c38331e2d287bc01
+        CONFIGURE_COMMAND sh -c "<SOURCE_DIR>/configure --prefix=<INSTALL_DIR> CXX=${CMAKE_CXX_COMPILER} CXXFLAGS='${CMAKE_CXX_FLAGS}' ${PACKAGE_COMPILER_FLAGS} ${PACKAGE_CONFIGURE_FLAGS}"
+#        GIT_REPOSITORY git://anongit.freedesktop.org/git/libreoffice/cppunit/
+#        GIT_TAG cppunit-1.13.2
+#        CONFIGURE_COMMAND sh -c "<SOURCE_DIR>/autogen.sh && <SOURCE_DIR>/configure --prefix=<INSTALL_DIR> CXX=${CMAKE_CXX_COMPILER} CXXFLAGS='${CMAKE_CXX_FLAGS}' ${PACKAGE_COMPILER_FLAGS} ${PACKAGE_CONFIGURE_FLAGS}"
+        BUILD_IN_SOURCE 1
+        BUILD_COMMAND make
+        INSTALL_COMMAND make install
+        LOG_DOWNLOAD ON
+        LOG_CONFIGURE ON
+        LOG_BUILD ON
+        )
+
+    ExternalProject_Get_Property(cppunit_external install_dir)
+    add_library(cppunit STATIC IMPORTED)
+    set_property(TARGET cppunit PROPERTY IMPORTED_LOCATION ${install_dir}/lib/libcppunit.a)
+endif()
+
 include_directories("../src")
 
 set(MainTest_SRCS
@@ -12,4 +39,11 @@ set(MainTest_SRCS
 )
 
 add_executable(MainTest ${MainTest_SRCS})
+
+if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
+    add_dependencies(MainTest cppunit_external)
+endif()
+
 target_link_libraries(MainTest cppunit airspace)
+
+# vim: expandtab

--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -1,4 +1,5 @@
-if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
+if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang" OR 
+        (NOT CPPUNIT AND NOT "${CMAKE_CXX_COMPILER_ID}" STREQUAL "MSVC"))
     # When building with Clang, we need to compile cppunit ourselves or we get
     # linking errors...
     include(ExternalProject)
@@ -25,25 +26,30 @@ if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
     set_property(TARGET cppunit PROPERTY IMPORTED_LOCATION ${install_dir}/lib/libcppunit.a)
 endif()
 
-include_directories("../src")
+if(CPPUNIT)
+    include_directories("../src")
 
-set(MainTest_SRCS
-    ArcTest.cpp
-    LatLonTest.cpp
-    LatitudeTest.cpp
-    LongitudeTest.cpp
-    CoordinateTest.cpp
-    MainTester.cpp
-    ParserTest.cpp
-    AirspaceTest.cpp
-)
+    set(MainTest_SRCS
+        ArcTest.cpp
+        LatLonTest.cpp
+        LatitudeTest.cpp
+        LongitudeTest.cpp
+        CoordinateTest.cpp
+        MainTester.cpp
+        ParserTest.cpp
+        AirspaceTest.cpp
+    )
 
-add_executable(MainTest ${MainTest_SRCS})
+    add_executable(MainTest ${MainTest_SRCS})
 
-if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
-    add_dependencies(MainTest cppunit_external)
+    if(TARGET cppunit_external)
+        add_dependencies(MainTest cppunit_external)
+    endif()
+
+    target_link_libraries(MainTest cppunit airspace)
+else()
+  message(WARNING "Cppunit was not found and we could not compile it "
+                  "automatically ourselves. Unit testing will be disabled.")
 endif()
-
-target_link_libraries(MainTest cppunit airspace)
 
 # vim: expandtab

--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -19,6 +19,7 @@ if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang" OR
         LOG_DOWNLOAD ON
         LOG_CONFIGURE ON
         LOG_BUILD ON
+        LOG_INSTALL ON
         )
 
     ExternalProject_Get_Property(cppunit_external install_dir)

--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -1,6 +1,6 @@
-if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
-    # When building with Clang, we need to compile cppunit ourselves or we get
-    # linking errors...
+if(NOT CPPUNIT OR "${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
+    # When cppunit was not found, or when building with Clang, we need to
+    # compile cppunit ourselves.
     include(ExternalProject)
     ExternalProject_Add(
         cppunit_external

--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -24,6 +24,7 @@ if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang" OR
     ExternalProject_Get_Property(cppunit_external install_dir)
     add_library(cppunit STATIC IMPORTED)
     set_property(TARGET cppunit PROPERTY IMPORTED_LOCATION ${install_dir}/lib/libcppunit.a)
+    include_directories("${install_dir}/include")
 
     # Flag that cppunit will be available.
     set(CPPUNIT True)

--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -24,6 +24,9 @@ if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang" OR
     ExternalProject_Get_Property(cppunit_external install_dir)
     add_library(cppunit STATIC IMPORTED)
     set_property(TARGET cppunit PROPERTY IMPORTED_LOCATION ${install_dir}/lib/libcppunit.a)
+
+    # Flag that cppunit will be available.
+    set(CPPUNIT True)
 endif()
 
 if(CPPUNIT)

--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -1,6 +1,6 @@
-if(NOT CPPUNIT OR "${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
-    # When cppunit was not found, or when building with Clang, we need to
-    # compile cppunit ourselves.
+if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
+    # When building with Clang, we need to compile cppunit ourselves or we get
+    # linking errors...
     include(ExternalProject)
     ExternalProject_Add(
         cppunit_external

--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -25,7 +25,10 @@ if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang" OR
     ExternalProject_Get_Property(cppunit_external install_dir)
     add_library(cppunit STATIC IMPORTED)
     set_property(TARGET cppunit PROPERTY IMPORTED_LOCATION ${install_dir}/lib/libcppunit.a)
-    include_directories("${install_dir}/include")
+    # Make sure to put SYSTEM in the following command! It will make the
+    # compiler consider the cppunit headers system headers, which makes them
+    # immune against certain warnings. See http://stackoverflow.com/questions/3371127/use-isystem-instead-of-i-with-cmake
+    include_directories(SYSTEM "${install_dir}/include")
 
     # Flag that cppunit will be available.
     set(CPPUNIT True)

